### PR TITLE
(DOCSP-20716) Documents telemetry for Atlas CLI

### PR DESCRIPTION
**Pull request by sarahsimpers**
[Build Link](https://workerpool-boxgs.mongodbstitch.com/pages/job.html?collName=queue&jobId=6287c22855f596c31d74e271)
[Jira Ticket Link](https://jira.mongodb.org/browse/DOCSP-20716)
[Staging](https://docs-mongodbcom-staging.corp.mongodb.com/atlas-cli/docsworker-xlarge/DOCSP-20716/telemetry/)

### CODE REVIEW

**Review by corryroot — APPROVED** [review]
LGTM! Just left some nits.

**Review by blva — APPROVED** [review]
LGTM, but would wait for @fmenezes final look!

File: source/atlas-cli-env-variables.txt [hunk 1]
```diff
@@ -98,4 +98,8 @@ The {+atlas-cli+} supports the following environment variables:
 
    * - ``MONGODB_ATLAS_REFRESH_TOKEN``
      - String that allows |service| to automatically request a new
-       access token when the current access token expires.
\ No newline at end of file
+       access token when the current access token expires.
+
+   * - ``MONGODB_ATLAS_TELEMETRY_ENABLED``
```
Comment (fmenezes) [inline source/atlas-cli-env-variables.txt:103]: I'm not too sure if we should document it but we allow users to set `DO_NOT_TRACK=1` to disable it as well

according to https://consoledonottrack.com/

File: source/telemetry.txt [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+   * - Operating System (OS) and OS version
+     - ``Windows 11.5``
+
+   * - Authentication method. {+atlas-cli+} telemetry does *not* track
+       the values for the |api| keys and login credentials.
+     - ``API key``   
+
+   * - Details for commands you run. {+atlas-cli+} telemetry tracks 
+       arguments only if they use pre-set allowable values, such as
```
Comment (fmenezes) [inline source/telemetry.txt:47]: we don't track values only which flags where sent, in your example it would be `[ "provider", "region" ]`

**Review by fmenezes — APPROVED** [review]

**Pull request merged**